torch/csrc/utils/tensor_new.cpp

```diff
@@ -453,19 +455,32 @@ Tensor internal_new_from_data(
     // "no observable data dependence".  In an ideal world, we wouldn't trace
     // a to() call but I need to think harder about what exactly we should trace
     // in this case.
-    tensor = tensor.to(
-        device, inferred_scalar_type, /*non_blocking=*/false, /*copy=*/false);
+    if (!get_lift_then_h2d()) {
+      tensor = tensor.to(
+          device, inferred_scalar_type, /*non_blocking=*/false, /*copy=*/false);
```

Comment:
Similar to my comment in `fake_tensor.py`, but: do we care about const-prop of dtype conversions still happening when this config is on? IIUC, we will still want to unconditionally do a `tensor.to(inferred_scalar_type)` to get that to work

Reply:
hmm, the config is supposed to change the ordering of `lift_fresh` w.r.t. to the tensor.to() call. Having something more fine-grained ("only do this for devices") adds more complexity to the code then I'd like ("lift then .to()" vs ".to() then lift" is easier to reason about).

If this is really a problem then we can make it more fine-grained in the future. The current solution doesn't prevent us from doing that.

I'll change the name of the config from `lift_then_h2d` to `lift_then_to` to better reflect the semantics

Reply:
sgtm (agreed it is extra complexity, and I don't have any real signal on how important const-propping dtype conversions is)